Soccar [Main-Server]/src/nl/soccar/mainserver/rmi/MainServerController.java

```diff
@@ -73,12 +81,54 @@ public MainServerController() {
         } catch (RemoteException e) {
             LOGGER.log(Level.SEVERE, "An error occurred while locating and/or binding the registry.", e);
         }
+
+        continuouslyPingGameServers();
+    }
+
+    /**
+     * Continously pings all game servers to check if the RMI connection is
+     * active. A random value is sent to the game servers for them to process
+     * with a pre-defined formula. If the result of the formula is not correct,
+     * the connected game server is possibly rogue. If the result of the formula
+     * is not correct or there was no response at all, the game server is
+     * deregistered.
+     */
+    public void continuouslyPingGameServers() {
+        timer.scheduleAtFixedRate(new TimerTask() {
+            @Override
+            public void run() {
+                synchronized (gameServers) {
+                    List<IGameServerForMainServer> serversToRemove = new ArrayList<>();
+
+                    gameServers.forEach(server -> {
+                        try {
+                            int pingCalculationValue = RANDOM.nextInt(100);
+                            if (pingCalculationValue * RmiConstants.PING_CALCULATION_FACTOR != server.ping(pingCalculationValue)) {
+                                throw new RogueGameServerException("Error", "The connected game server did not return the correct result of the predefined ping formula.");
+                            }
+                        } catch (RemoteException e) {
+                            serversToRemove.add(server);
+
+                            LOGGER.log(Level.WARNING, "A game server is deregistered because it is no longer active.", e);
+                        } catch (RogueGameServerException e) {
+                            serversToRemove.add(server);
+
+                            LOGGER.log(Level.WARNING, "A game server is deregisterd because it did not return the correct result of the predefined ping formula.", e);
+                        }
+                    });
+
+                    serversToRemove.stream().forEach(s -> deregisterGameServer(s));

```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1612)